### DISCUSSION TITLE
Android port?

### DISCUSSION THREAD

**Discussion by Malaska678** [oracle/graal#2822] [General] [answered]
Actually there's open jdk port but i think graal port can be better.
It benifits porters who port pc-only java games to android (Like minecraft java edition or the pc only version)

**Comment by Malaska678** [comment]
Want it to support all android 5+ and above

**Comment by Malaska678** [comment]
Accident

**Comment by munishchouhan** [comment]
@Malaska678 Please provide the detailed information of this feature in the below format:

**1. Is your feature request related to a problem? Please describe.**
**2. Describe the solution you'd like.**
**3. Describe who do you think will benefit the most.**
**4. Describe alternatives you've considered.**
**5. Additional context.**

**Reply by Malaska678** [reply]
1.) Yup. Porting a pc java game to android sometimes need jdk port for android. There open jdk for android but graalvm for android can be better.
2.) Want GraalVM android port. Work for android 5+
3.) PC Java games porter to android
4.) Nothing.
5.) We want graalVM port for android port 👍

**Reply by Malaska678** [reply]
So why u turn this to a discussion

**Reply by Malaska678** [reply]
Hey @mcraj017 , why no one not a single developer port graal vm to android. Why?!

**Comment by dougxc** [comment]
I converted this to a discussion as we already have a number of open issues related to porting GraalVM to Android. The issue you opened didn't add anything new so I felt it would be better as a discussion for others to join in terms of voicing support for such a port.

In terms of your question "why no one not a single developer port graal vm to android", this is an open source project and contributions are always welcome.

**Comment by vjovanov** [comment, marked as answer]
We have the Android support ready via the GraalVM Native Image. For instructions on how to do that, see 

  https://github.com/gluonhq/client-samples

Porting games is indeed an interesting topic. I believe Minecraft Java Edition could be ported to both iOS and Android with a reasonable effort.

**Reply by Malaska678** [reply]
Im not the developer. But @khanhduytran0 Is the developer.

**Reply by Malaska678** [reply]
Did u think graalvm can be better than openjdk android port?

**Reply by vjovanov** [reply]
Depends if you use the JIT or not on Android. If you use the JIT you will get better performance than GraalVM CE. However, your application will be way larger, consume more memory, and also have a significantly worse startup.

With GraalVM EE you will get better performance than the port of Open JDK with better memory footprint and startup.

**Reply by Malaska678** [reply]
Graal vm ee? Is it paid?

**Reply by Malaska678** [reply]
So if u want to try mcinabox(a minecraft JE for android) with graal vm Then u need create graalvm runtime(mcinabox use open jdk runtime).Do this procedure to Graal vm CE instead of doing it on open jdk

**Reply by Malaska678** [reply]
![Screenshot_2020-09-03-22-11-18-10](https://user-images.githubusercontent.com/68141607/92126542-e2daff80-ee32-11ea-90a8-35f27485199b.jpg)

**Reply by Malaska678** [reply]
And make it support all android device not just aarch 64. Also 1.13+ may not work

**Reply by Malaska678** [reply]
Read all text above

**Reply by vjovanov** [reply]
> Graal vm ee? Is it paid?

For mobile, it is not yet defined.

**Reply by vjovanov** [reply]
You would still need to compile the native libs for AARCH64. But the rest should work out.

**Reply by vjovanov** [reply]
> And make it support all android device not just aarch 64. Also 1.13+ may not work

We could support that if necessary on the GraalVM side.

**Reply by Malaska678** [reply]
Thanks

**Reply by khanhduytran0** [reply]
If we use JIT then it would not possible to bring to iOS (althought iOS 14.2+ with Apple A12+ allows JIT, but when will GraalVM JIT being ported to iOS?)

If we use AOT native-image then another problem: Mojang does not allow redistribute game since Minecraft JE is a paid game.

**Comment by Malaska678** [comment]
I tried to do same on openj9 github but developer response was a lot slower.

**Reply by bobvandette** [reply]
Doesn't Gluon already support GraalVM & JavaFX on Android?
https://github.com/gluonhq/client-samples

**Reply by vjovanov** [reply]
Yes, that is what we are talking about.

**Reply by khanhduytran0** [reply]
But then how a GLES context can be obtained from JavaFX?

**Comment by SavagePy10** [comment]
My old cringy question!!!! Hate this....
